contracts/BaseStrategy.sol

```diff
@@ -847,6 +846,25 @@ abstract contract BaseStrategy {
 
         IERC20(_token).safeTransfer(governance(), IERC20(_token).balanceOf(address(this)));
     }
+
+    function liquidateEverything(uint256 debtOutstanding)
+        internal
+        returns (
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        uint256 loss = 0;
+        uint256 profit = 0;
+        uint256 amountFreed = liquidateEverything();
+        if (amountFreed < debtOutstanding) {
+            loss = debtOutstanding.sub(amountFreed);
+        } else if (amountFreed > debtOutstanding) {
+            profit = amountFreed.sub(debtOutstanding);
+        }
+        return (profit, loss, debtOutstanding.sub(loss));
```

Comment:
Can this be integrated into `harvest` above? I think it 's a lot clearer than this one degree of indirection, especially with the naming of this function

Reply:
Yes